posthog/clickhouse/test/test_person_overrides.py

```diff
@@ -0,0 +1,87 @@
+import json
+from datetime import datetime, timedelta
+from time import sleep
+from typing import List, cast
+from uuid import uuid4
+
+import pytest
+import pytz
+from kafka import KafkaProducer
+
+from posthog.clickhouse.client import sync_execute
+from posthog.kafka_client.topics import KAFKA_PERSON_OVERRIDE
+from posthog.models.person_overrides.sql import (
+    DROP_KAFKA_PERSON_OVERRIDES_TABLE_SQL,
+    DROP_PERSON_OVERRIDES_CREATE_MATERIALIZED_VIEW_SQL,
+    KAFKA_PERSON_OVERRIDES_TABLE_SQL,
+    PERSON_OVERRIDES_CREATE_MATERIALIZED_VIEW_SQL,
+)
+from posthog.settings.data_stores import KAFKA_HOSTS
+
+
+@pytest.mark.django_db
+def test_can_insert_person_overrides():
+    # By default the test suite runs with ClickHouse no Kafka or Materialixzed
+    # Views. Updating to include these seems to be a [larger task to
+    # fix](https://github.com/PostHog/posthog/pull/13878), so here we create
+    # just the missing tables we need to verify functionality.
+    sync_execute(KAFKA_PERSON_OVERRIDES_TABLE_SQL)
+    sync_execute(PERSON_OVERRIDES_CREATE_MATERIALIZED_VIEW_SQL)
+
+    producer = KafkaProducer(bootstrap_servers=KAFKA_HOSTS)
+    try:
```

Comment:
nit: I don't like wrapping everything in a try/finally. It works, but:
* It adds a level of indentation
* It's not clear which of the code could fail 

Pytest has some nice tools to do exactly this in [`pytest.fixtures`](https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended) and the [`addfinalizer`](https://docs.pytest.org/en/6.2.x/fixture.html#adding-finalizers-directly) method, which were built to run cleanup operations like `producer.close()`.

But it's not a big deal, ultimately it's purely about read-ability. We can do a pass later and clean up a lot of these tests which may re-use the same fixtures/finalizers.

Reply:
tbh I find the try/finally to be more obvious what is going on. I'm going to leave as is for now.